ddsenabler_participants/include/ddsenabler_participants/rpc/RpcStructs.hpp

```diff
@@ -0,0 +1,577 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file RpcStructs.hpp
+ */
+
+#pragma once
+
+
+#include <ddsenabler_participants/rpc/RpcTypes.hpp>
+#include <ddsenabler_participants/Constants.hpp>
+
+#include <ddspipe_core/types/dds/Endpoint.hpp>
+
+#include <optional>
+
+namespace eprosima {
+namespace ddsenabler {
+namespace participants {
+
+enum class ActionEraseReason
+{
+    FINAL_STATUS,  // Erase triggered by receiving final status (e.g., succeeded, aborted, canceled)
+    RESULT,        // Erase triggered by receiving result
+    FORCED         // Force erase regardless of status/result
+};
+
+struct RpcAction
+{
+    RpcAction() = default;
+    RpcAction(
+            const std::string& action_name,
+            const ddspipe::core::types::RpcTopic& goal,
+            const ddspipe::core::types::RpcTopic& result,
+            const ddspipe::core::types::RpcTopic& cancel,
+            const ddspipe::core::types::DdsTopic& feedback,
+            const ddspipe::core::types::DdsTopic& status)
+        : action_name(action_name)
+        , goal(goal)
+        , result(result)
+        , cancel(cancel)
+        , feedback(feedback)
+        , status(status)
+    {
+    }
+
+    std::string action_name;
+    ddspipe::core::types::RpcTopic goal;
+    ddspipe::core::types::RpcTopic result;
+    ddspipe::core::types::RpcTopic cancel;
+    ddspipe::core::types::DdsTopic feedback;
+    ddspipe::core::types::DdsTopic status;
+};
+
+struct RpcInfo
+{
+    RpcInfo(
+            const std::string& dds_topic_name)
+        : topic_name(dds_topic_name)
+        , rpc_protocol(RpcProtocol::PROTOCOL_UNKNOWN)
+        , rpc_type(RpcType::NONE)
+        , service_type(ServiceType::NONE)
+        , action_type(ActionType::NONE)
+    {
+        detect_rpc_protocol();
+        try
+        {
+            remove_prefix_suffix();
+        }
+        catch(const std::exception& e)
+        {
+            throw std::runtime_error("Error extracting RPC info from topic name '" + topic_name + "': " + e.what());
+        }
+    }
+
+    void detect_rpc_protocol()
+    {
+        if (topic_name.rfind(ROS_TOPIC_PREFIX, 0) == 0 ||
+                topic_name.rfind(ROS_REQUEST_PREFIX, 0) == 0 ||
+                topic_name.rfind(ROS_REPLY_PREFIX, 0) == 0)
+        {
+            rpc_protocol = RpcProtocol::ROS2;
+            return;
+        }
+        else if (topic_name.rfind(FASTDDS_TOPIC_PREFIX, 0) == 0 ||
+                topic_name.rfind(FASTDDS_REQUEST_PREFIX, 0) == 0 ||
+                topic_name.rfind(FASTDDS_REPLY_PREFIX, 0) == 0)
+        {
+            rpc_protocol = RpcProtocol::FASTDDS;
+            return;
+        }
+        rpc_protocol = RpcProtocol::PROTOCOL_UNKNOWN;
```

Comment:
Will this case ever be covered? Fast-DDS prefix "" will match with anything right? And maybe at some point it should be checked if the protocol is unknown and fail.